bundles/org.openhab.binding.fenecon/src/main/java/org/openhab/binding/fenecon/internal/FeneconHandler.java

```diff
@@ -0,0 +1,231 @@
+            try {
+                @SuppressWarnings("null")
+                HttpRequest request = baseHttpRequest
+                        .uri(new URI(getBaseUrl(config) + "rest/channel/_sum/" + eachChannel)).build();
+                logger.trace("FENECON - request: {}", request);
+
+                HttpResponse<String> response = httpClient.send(request, BodyHandlers.ofString());
+                logger.trace("FENECON - response status code: {} body: {}", response.statusCode(), response.body());
+
+                if (response.statusCode() > 300) {
+                    // Authentication error
+                    if (response.statusCode() == 401) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                                parseFeneconError(response));
+                        return;
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                                "http status code: " + response.statusCode());
+                        return;
+                    }
+                } else {
+                    processDataPoint(JsonParser.parseString(response.body()).getAsJsonObject());
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (URISyntaxException | JsonSyntaxException | IOException | InterruptedException err) {
+                logger.trace("FENECON - connection problem on FENECON channel {}", eachChannel, err);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, err.getMessage());
+                return;
+            }
+        }
+
+        // Set last successful update cycle
+        updateState("lastUpdate", new DateTimeType());
+    }
+
+    private void processDataPoint(JsonObject response) {
+        // Example: {"address":"_sum/EssSoc","type":"INTEGER","accessMode":"RO","text":"Range
+        // 0..100","unit":"%","value":99}
+        String address = response.get("address").getAsString();
+        String text = response.get("text").getAsString();
+        String value = response.get("value").getAsString();
+
+        switch (address) {
+            case "_sum/State":
+                // {"address":"_sum/State","type":"INTEGER","accessMode":"RO","text":"0:Ok, 1:Info, 2:Warning,
+                // 3:Fault","unit":"","value":0}
+                int begin = text.indexOf(value + ":");
+                int end = text.indexOf(",", begin);
+                updateState("state", new StringType(text.substring(begin + 2, end)));
+                break;
+            case "_sum/EssSoc":
+                updateState("essSoc", new QuantityType<>(Integer.valueOf(value), Units.PERCENT));
+                break;
+            case "_sum/ConsumptionActivePower":
+                updateState("consumptionActivePower", new QuantityType<>(Integer.valueOf(value), Units.WATT));
```

Comment:
See other comment, use constants for the channel id's

Reply:
Done